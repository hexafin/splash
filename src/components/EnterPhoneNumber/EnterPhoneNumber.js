import React, {Component} from "react"
import {
    View,
    Text,
    StyleSheet,
    Image,
    Alert,
    TouchableOpacity,
    KeyboardAvoidingView,
    Keyboard
} from "react-native"
import {Input} from "../universal/Input"
import FlatBackButton from "../universal/FlatBackButton"
import Button from "../universal/Button"
import {colors} from "../../lib/colors"
import {defaults, icons} from "../../lib/styles"
import PhoneNumberInput from "../universal/PhoneNumberInput"

class EnterPhoneNumber extends Component {

    constructor(props) {
        super(props)
        this.state = {
            phoneNumber: {
                countryName: "",
                countryFlag: "",
                countryCode: "",
                number: ""
            },
            isLoading: false
        }
    }

    componentWillMount() {
        if (this.props.splashtag == "") {
            this.props.navigation.navigate("ChooseSplashtag")
        }
    }

    componentDidUpdate(prevProps) {
        if (this.props.smsError != prevProps.smsError && this.props.smsError) {
          Alert.alert('An error occurred!', 'Sorry about this. Our team has been notified and we should fix this shortly!', [
              {
                  text: 'Cancel',
                  onPress: () => console.log('Cancel Pressed'),
                  style: 'cancel'
              }, {
                  text: 'Ok',
                  onPress: () => console.log('OK Pressed')
              }
          ])
        }
    }

    render() {
        return (<KeyboardAvoidingView style={styles.container} behavior={"height"}>

            <View style={styles.body}>

                <Text style={styles.title}>
                    Welcome,{"\n"}
                    @{this.props.splashtag}
                </Text>

                <Text style={styles.subtitle}>
                    Verify your number to claim{"\n"}
                    your splashtag
                </Text>

                <PhoneNumberInput autoFocus={true} callback={(phoneNumber) => {
                        this.setState((prevState) => {
                            return {
                                ...prevState,
                                phoneNumber
                            }
                        })
                    }}/>

                <Text style={styles.description}>
                    We{"'"}ll text you a verification code{"\n"}
                    to make sure it{"'"}s you
                </Text>

                <Button onPress={() => {

                        // remove spaces from phone number
                        var fullNumber = this.state.phoneNumber.countryCode + this.state.phoneNumber.number
                        fullNumber = fullNumber.replace(/\s/g, '')

                        // initiate animation
                        this.setState((prevState) => {
                            return {
                                ...prevState,
                                isLoading: true
                            }
                        })

                        // intitiate authentication process
                        this.props.SmsAuthenticate(fullNumber, this.state.phoneNumber.countryName)
                        Keyboard.dismiss()

                    }} style={styles.footerButton}
                    title={"Text me the code"}
                    primary={true} loading={true}
                    loading={this.state.isLoading}
                    disabled={this.state.phoneNumber.number.length < 12}/>

            </View>

            <FlatBackButton onPress={() => {
                    // Keyboard.dismiss()
                    this.props.navigation.navigate("ChooseSplashtag")
                }}/>
        </KeyboardAvoidingView>)

    }

}

const styles = StyleSheet.create({
    container: {
        ...defaults.container,
        justifyContent: "space-between",
        position: "relative",
        paddingBottom: 0,
        paddingTop: 80
    },
    body: {
        flex: 1,
        padding: 20,
        flexDirection: "column",
        justifyContent: "space-between"
    },
    title: {
        fontSize: 30,
        fontWeight: '500',
        color: colors.nearBlack,
        marginBottom: 20,
        textAlign: "center"
    },
    subtitle: {
        fontSize: 20,
        textAlign: "center",
        color: colors.nearBlack,
        marginBottom: 20,
        zIndex: 10
    },
    description: {
        fontSize: 20,
        textAlign: "center",
        color: colors.lightGray,
        marginBottom: 20,
        marginTop: 15,
        zIndex: 40
    }
})

export default EnterPhoneNumber
