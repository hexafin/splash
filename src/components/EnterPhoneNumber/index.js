import EnterPhoneNumber from "./EnterPhoneNumber"
import {connect} from "react-redux"
import {bindActionCreators} from "redux"
import {ClaimUsername, InviteFriends} from "../../actions/waitlist"

const mapStateToProps = (state) => {

    let splashtag = ""
    if (typeof state.form.chooseSplashtag !== 'undefined' && state.form.chooseSplashtag.values) {
        splashtag = state.form.chooseSplashtag.values.splashtag
    }

    let phoneNumber = ""
    if (typeof state.form.enterPhoneNumber !== 'undefined' && state.form.enterPhoneNumber.values) {
        phoneNumber = state.form.enterPhoneNumber.values.phoneNumber
    }

    return {
      phoneNumber: phoneNumber,
      splashtag: splashtag
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        ClaimUsername, InviteFriends
    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(EnterPhoneNumber)
