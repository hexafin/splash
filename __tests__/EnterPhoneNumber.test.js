import "react-native";
import React from "react";
import EnterPhoneNumber from "../src/components/EnterPhoneNumber/EnterPhoneNumber";

it("Enter Phone Number renders correctly", () => {
	const snap = shallow(<EnterPhoneNumber />);
	expect(snap).toMatchSnapshot();
});
