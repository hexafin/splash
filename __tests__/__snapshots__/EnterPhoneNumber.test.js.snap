// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enter Phone Number renders correctly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingBottom": 0,
      "paddingTop": 80,
      "position": "relative",
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#333333",
          "fontSize": 29,
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
    >
      Welcome,
      

    </Component>
    <Component
      style={
        Object {
          "color": "#333333",
          "fontSize": 17,
          "marginBottom": 20,
          "textAlign": "center",
          "zIndex": 10,
        }
      }
    >
      Verify your number to 
      recover
      

      your wallet
    </Component>
    <PhoneNumberInput
      autoFocus={true}
      callback={[Function]}
      number=""
    />
    <Component
      style={
        Object {
          "color": "#ADADAD",
          "fontSize": 20,
          "marginBottom": 20,
          "marginTop": 15,
          "textAlign": "center",
          "zIndex": 40,
        }
      }
    >
      We
      '
      ll text you a verification code
      

      to make sure it
      '
      s you
    </Component>
    <Button
      disabled={true}
      loading={false}
      onPress={[Function]}
      primary={true}
      title="Text me the code"
    />
  </Component>
  <FlatBackButton
    onPress={[Function]}
  />
</KeyboardAvoidingView>
`;
